pkg/cgroups/cgroups.go

```diff
@@ -92,8 +92,15 @@ func getCgroupRootForKind() (string, error) {
 			return "", fmt.Errorf("cgroups v1 net_cls or net_prio detected which interferes with cgroup v2")
 		}
 
+		// Inputs : (nice to have unit tests to validate the supported inputs)
+		// line := "0::/system.slice/docker-$(SOME_NODE_ID).scope/kubelet/kubepods/burstable/pod-$(SOME_POD_ID)"
+		// line := "0::/kubelet/kubepods/burstable/pod95f45c87-c016-4b1a-9692-73ae901ea461/de0c46065bb736b5e0b8b85e8fec48b3cdee83351418552e9af300fd64ed4cbe"
+		// line := "0::/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod6745a7f5_63fd_46ff_8ce6_e312ac764642.slice/docker-0c2c12c9762dbe9f3e5c3bc43224347b828d19bd1834ed6a6d87fbd3d7bbb4e4.scope"
+		//
+		// Why kubepods or kubepods.slice?
+		// cgroupfs vs systemd cgroup driver -- Reference : https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cgroup_manager_linux.go#L86
 		if strings.HasPrefix(line, "0::") { // cgroup v2 starts with "0::"
-			re := regexp.MustCompile(`^.*kubelet`)
+			re := regexp.MustCompile(`^.*(kubepods|kubepods.slice)/`)
 			p := path.Join(cgroupRoot, strings.Split(line, "::")[1])
```

Comment:
Thanks for the patch and digging into kubelet code! :)

How about we leave detection logic outside of the cilium agent, and just pass a proper argument to `--cgroup-root` in the agent? We can even pass two options, one for the root and one for where to attach the bpf programs, if we care about this distinction. We already had a special case for Kind, now we are going to add another one, and I think we should keep this complexity out of the agent.